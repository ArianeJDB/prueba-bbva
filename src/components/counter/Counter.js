import React, { useState } from 'react';


const Counter = () => {
    return(
        <div>
            counter is working
        </div>
    )
}

export default Counter;