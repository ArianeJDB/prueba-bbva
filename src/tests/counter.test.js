import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import Counter from '../components/counter/Counter'

describe('Counter component', () => {
    let wrapper;
    test('should render', () => {
        wrapper = shallow(<Counter params={{
            match: 
            {isExact: true,
            params: {
                email: "jax@jax.com"
            },
                path: "/counter/:email",
                url: "/counter/jax@jax.com"}
            }} />);
        expect(wrapper).toMatchSnapshot()
    })
    test('should render h1 with Welcome message', () => {
        wrapper = shallow(<Counter params={{
            match: 
            {isExact: true,
            params: {
                email: "jax@jax.com"
            },
                path: "/counter/:email",
                url: "/counter/jax@jax.com"}
            }} />);
        expect(wrapper.containsMatchingElement(<h1>Welcome!</h1>)).toBeTruthy()
        expect(wrapper.find('.counter_container').exists()).toBe(true)
    })
})
