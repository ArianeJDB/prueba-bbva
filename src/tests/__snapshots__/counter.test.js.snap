// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component should render 1`] = `
<div
  className="counter_section"
>
  <h1>
    Welcome!
  </h1>
  <p>
    The last time you accessed was
  </p>
  <div
    className="counter_container"
  >
    <div
      className="number"
    >
      0
       
      <span>
        days
      </span>
    </div>
    <div
      className="number"
    >
      0
       
      <span>
        hours
      </span>
    </div>
    <div
      className="number"
    >
      0
       
      <span>
        minutes
      </span>
    </div>
    <div
      className="number"
    >
      0
       
      <span>
        seconds
      </span>
    </div>
  </div>
  <button
    className="logout-btn"
    onClick={[Function]}
  >
    LOGOUT
  </button>
</div>
`;
